### ISSUE TITLE
Webp breaks the layout of a product in the shop

### ISSUE THREAD

**Issue by anselmpaul** [pretix/pretix#4308]
### Problem and impact

When using webp files as product image, the picture is rendered in full size, breaking out of the layout, creating a weird text flow and also leading to a buggy UI in some layout sizes

### Expected behaviour

When viewing products in the shop, there is a small product picture preview which can be enlarged on click. This works fine for jpgs or pngs, being rendered at 60x60px

### Steps to reproduce

1. Add a product with an webp image to your shop, add a title and description
2. Open the shop

### Screenshots

![image](https://github.com/user-attachments/assets/d42f9baa-6340-493e-968b-62f32a1fd989)

![image](https://github.com/user-attachments/assets/b166c330-a906-435b-998e-a482ff495d81)

### Browser (software, desktop or mobile?) and version

Chrome on Desktop

### Operating system, dependency versions

Macos

### Version

hosted at pretix.eu

**Comment by anselmpaul** [comment]
Sure thing. I had to zip it since github wouldn't let me attach it as webp.
[crashing.webp.zip](https://github.com/user-attachments/files/16264627/crashing.webp.zip)

**Comment by wiffbi** [comment]
It’s simply that `WEBP` is not in the list of supported image-formats in `PILLOW_FORMATS_QUESTIONS_IMAGE`. Locally it worked for me to just add it to it, but it might need some updates on the server as well, so das webp is supported.

Also maybe we should limit the allowed images in the file-input accept-attribute? Currently it is `image/*`, but we can basically handle only `('PNG', 'GIF', 'JPEG', 'BMP', 'TIFF')`

**Comment by raphaelm** [comment]
How was this ever allowed to be uploaded then :thinking:

**Comment by wiffbi** [comment]
Well, I guess the default django `ImageField` just validates if it is any image at all (or probably an image, that Pillow can handle). But when we generate the preview, we use the `formats` parameter with `PILLOW_FORMATS_QUESTIONS_IMAGE` in Image.open, which then fails on webp-images and just passes them through directly. I am not wure, whether we should just let Pillow try and handle any image when creating the thumbnail or limit the validation to the ones allowed in `PILLOW_FORMATS_QUESTIONS_IMAGE` – and then extend to webp as well?

**Comment by raphaelm** [comment]
I believe the first step is to bring this in line with all the other fields in the system where we explicitly check the extension. Then, we can think about adding webp as an additional extension?

**Comment by raphaelm** [comment]
> just let PIllow try to handle anything

no, absolutely not, we introduced these format limitations specifically due to pillow security issues e.g. with EPS

**Comment by wiffbi** [comment]
#4316 fixes that one could upload any image for a product picture and limits it to the existing list of png, jpg, etc. It does not add webp-support though. AFAICT to add webp-support, we need to make sure, that Pillow has webp-support, which depends on the server-setup. Furthermore, if we add it to the `PILLOW_FORMATS_QUESTIONS_IMAGE` we need to check, whether we use those uploaded images somewhere other than web (e.g. PDF-Tickets, etc.) and need to check whether webp is supported there as well. So as easy it sounds to add webp to `PILLOW_FORMATS_QUESTIONS_IMAGE`, there has to be done a lot of testing. So for now, #4316 just fixes that one could upload webp as a product image at all.

Sorry that I do not have better news regarding webp-support.
